executor_manager/routers/routers.py

```
@@ -172,6 +172,220 @@ class CancelTaskRequest(BaseModel):
+            valid=False, checks=[], errors=[f"Unknown shell type: {shell_type}"]
+        )
+
+    checks_to_run = checks_config[shell_type]
+    results = []
+    errors = []
+    all_passed = True
+
+    try:
+        # Pull the image first (with timeout)
+        logger.info(f"Pulling image {image} for validation...")
+        pull_result = subprocess.run(
```

Ruff (0.14.6)

`subprocess` call: check for execution of untrusted input

(S603)